teuthology/test/task/__init__.py

```diff
@@ -0,0 +1,132 @@
+from mock import patch
+from pytest import raises
+
+from teuthology.config import FakeNamespace
+from teuthology.orchestra.cluster import Cluster
+from teuthology.orchestra.remote import Remote
+from teuthology.task import Task
+
+
+class TestTask(object):
+    def setup(self):
+        self.ctx = FakeNamespace()
+        self.ctx.config = dict()
+
+    def test_overrides(self):
+        self.ctx.config['overrides'] = dict(
+            task=dict(
+                key_1='overridden',
+            ),
+        )
+        task_config = dict(
+            key_1='default',
+            key_2='default',
+        )
+        with Task(self.ctx, task_config) as task:
+            assert task.config['key_1'] == 'overridden'
+            assert task.config['key_2'] == 'default'
+
+    def test_hosts_no_results(self):
+        self.ctx.cluster = Cluster()
+        self.ctx.cluster.add(Remote('remote1'), ['role1'])
+        task_config = dict(
+            hosts=['role2'],
+        )
+        with raises(RuntimeError):
+            with Task(self.ctx, task_config) as task:
+                task_hosts = task.hosts
+                assert len(task_hosts) == 0
+
+    def test_hosts_one_role(self):
+        self.ctx.cluster = Cluster()
+        self.ctx.cluster.add(Remote('remote1'), ['role1'])
+        self.ctx.cluster.add(Remote('remote2'), ['role2'])
+        task_config = dict(
+            hosts=['role1'],
+        )
+        with Task(self.ctx, task_config) as task:
+            task_hosts = task.hosts
+            assert len(task_hosts) == 1
+            assert task_hosts[0].name == 'remote1'
+
+    def test_hosts_two_roles(self):
+        self.ctx.cluster = Cluster()
+        self.ctx.cluster.add(Remote('remote1'), ['role1'])
+        self.ctx.cluster.add(Remote('remote2'), ['role2'])
+        self.ctx.cluster.add(Remote('remote3'), ['role3'])
+        task_config = dict(
+            hosts=['role1', 'role3'],
+        )
+        with Task(self.ctx, task_config) as task:
+            task_hosts = task.hosts
+            assert len(task_hosts) == 2
+            hostnames = [host.name for host in task_hosts]
+            assert sorted(hostnames) == ['remote1', 'remote3']
+
+    def test_hosts_two_hostnames(self):
+        self.ctx.cluster = Cluster()
+        self.ctx.cluster.add(Remote('remote1.example.com'), ['role1'])
+        self.ctx.cluster.add(Remote('remote2.example.com'), ['role2'])
+        self.ctx.cluster.add(Remote('remote3.example.com'), ['role3'])
+        task_config = dict(
+            hosts=['remote1', 'remote2.example.com'],
```

Comment:
Yeah, I'm not sure if that will ever be useful in practice, but it was easy to implement and test.